coffeecard/CoffeeCard.Library/Services/v2/TicketService.cs

```diff
@@ -83,6 +86,35 @@ public async Task<UsedTicketResponse> UseTicketAsync(User user, int productId)
             };
         }
 
+        public async Task<UsedTicketResponse> UseTicketAsync(User user, int productId, int menuItemId)
```

Comment:
I believe it would be good to check to check if the ProductId is valid (that it exists) at first in the method before querying for tickets or menu items

Reply:
Addressed in 4011699489e03cefe6b6666037d7c3262bc97abd